oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -1755,53 +1755,178 @@ int mca_spml_ucx_team_sync(shmem_team_t team)
         shmem_team_config_t *config)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_team = (mca_spml_ucx_team_t *)team;
+    SPML_UCX_VALIDATE_TEAM(team);
+
+    memcpy(config, &ucx_team->config, sizeof(shmem_team_config_t));
+
+    return SHMEM_SUCCESS;
+}
+
+static inline int mca_spml_ucx_is_pe_in_strided_team(int src_pe, int start,
+        int stride, int size)
+{
+    return (src_pe >= start) && (src_pe < start + size * stride)
+           && ((src_pe - start) % stride == 0);
 }
 
-/* This routine is not implemented */
 int mca_spml_ucx_team_translate_pe(shmem_team_t src_team, int src_pe,
         shmem_team_t dest_team)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_src_team  = (mca_spml_ucx_team_t*) src_team;
+    mca_spml_ucx_team_t *ucx_dest_team = (mca_spml_ucx_team_t*) dest_team;
+    int global_pe;
+
+    if ((src_pe == SPML_UCX_PE_NOT_IN_TEAM) || (src_team == dest_team)) {
+        return src_pe;
+    }
+
+    global_pe = ucx_src_team->start + src_pe * ucx_src_team->stride;
+
+    if (dest_team == SHMEM_TEAM_WORLD) {
+        return global_pe;
+    }
+
+    if (!mca_spml_ucx_is_pe_in_strided_team(global_pe, ucx_dest_team->start, ucx_dest_team->stride,
+                                            ucx_dest_team->n_pes)) {
+        return SPML_UCX_PE_NOT_IN_TEAM;
+    }
+
+    return (global_pe - ucx_dest_team->start) / ucx_dest_team->stride;
 }
 
-/* This routine is not implemented */
 int mca_spml_ucx_team_split_strided(shmem_team_t parent_team, int start, int
         stride, int size, const shmem_team_config_t *config, long config_mask,
         shmem_team_t *new_team)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_parent_team;
+    mca_spml_ucx_team_t *ucx_new_team;
+    int parent_pe;
+    int parent_start;
+    int parent_stride;
+    int my_pe;
+
+    SPML_UCX_ASSERT(((start + size * stride) <= oshmem_num_procs()) && (start < size) && (stride > 0) && (size > 0));
+
+    if (parent_team == SHMEM_TEAM_WORLD) {
+        parent_pe = shmem_my_pe();
+        parent_start = 0;
+        parent_stride = 1;
+    } else {
+        ucx_parent_team = (mca_spml_ucx_team_t*) parent_team;
+        parent_pe = ucx_parent_team->my_pe;
+        parent_start = ucx_parent_team->start;
+        parent_stride = ucx_parent_team->stride;
+    }
+
+    if (mca_spml_ucx_is_pe_in_strided_team(parent_pe, start, stride, size)) {
+        my_pe = (parent_pe - start) / stride;
+    } else {
+        /* not in team, according to spec it should be SHMEM_TEAM_INVALID but its value is NULL which
+        can be also interpreted as 0 (first pe), therefore -1 is used */
+        my_pe = SPML_UCX_PE_NOT_IN_TEAM;
+    }
+
+    /* In order to simplify pe translations start and stride are calculated with respect to
+     * world_team */
+    ucx_new_team = (mca_spml_ucx_team_t *)malloc(sizeof(mca_spml_ucx_team_t));
+    ucx_new_team->start = parent_start + start;
```

Comment:
this is incorrect, we should consider the parent's stride as well when we want to calculate the new stride based on the comm world:
```ucx_new_team->start = parent_start + (start * parent_stride)```